packages/react-devtools-shared/src/devtools/utils.js

```diff
@@ -21,10 +26,25 @@ export function printElement(element: Element, includeWeight: boolean = false) {
     key = ` key="${element.key}"`;
   }
 
-  let hocs = '';
+  let hocDisplayNames = null;
   if (element.hocDisplayNames !== null) {
-    hocs = ` [${element.hocDisplayNames.join('][')}]`;
+    hocDisplayNames = [...element.hocDisplayNames];
   }
+  if (element.type === ElementTypeMemo) {
+    if (hocDisplayNames === null) {
+      hocDisplayNames = ['Memo'];
+    } else {
+      hocDisplayNames.push('Memo');
+    }
+  } else if (element.type === ElementTypeForwardRef) {
+    if (hocDisplayNames === null) {
+      hocDisplayNames = ['ForwardRef'];
+    } else {
+      hocDisplayNames.push('ForwardRef');
+    }
+  }
+
+  let hocs = hocDisplayNames === null ? '' : ` [${hocDisplayNames.join('][')}]`;
```

Comment:
Serialize the store in a format that includes the HOC badges, more like the `Badge` component does in the real DevTools UI. This will make our tests more meaningful.

Reply:
Yeah this is pretty. Nice.